aptos-move/framework/aptos-framework/sources/TokenCoinSwap.move

```diff
@@ -0,0 +1,231 @@
+// A basic utility module for swapping a fixed amount of coins with a fixed amount of tokens.
+module AptosFramework::TokenCoinSwap {
+    use Std::Signer;
+    use Std::Option::Option;
+    use AptosFramework::Table::{Self, Table};
+    use AptosFramework::Token::{Self, Token, TokenId};
+    use AptosFramework::Coin::{Self, Coin};
+    use AptosFramework::SimpleMap::{SimpleMap, Self};
+    use AptosFramework::Timestamp;
+
+    const ETOKEN_ALREADY_LISTED: u64 = 1;
+    const ETOKEN_LISTING_NOT_EXIST: u64 = 2;
+    const ETOKEN_NOT_IN_ESCROW: u64 = 3;
+    const ETOKEN_CANNOT_MOVE_OUT_OF_ESCROW_BEFORE_LOCKUP_TIME: u64 = 4;
+    const ETOKEN_MIN_PRICE_NOT_MATCH: u64 = 5;
+    const ETOKEN_AMOUNT_NOT_MATCH: u64 = 6;
+    const ENOT_ENOUGH_COIN: u64 = 7;
+
+    struct TokenCoinSwap<phantom CoinType> has store, drop {
+        token_amount: u64,
+        min_price_per_token: u64,
+    }
+
+    // stored at token owner's account
```

Comment:
Suggest to use `///` and proper english. We will have documentation generator running again.

All our Move code should be thoroughly documented; every public entity should have documentation comment.

Reply:
updated